### PR TITLE
fix: add unit test for assets polling loops

### PR DESCRIPTION
**Pull request by salimtb**
## **Description**
the goal of this PR is to add unit tests for all multichain assets polling loops

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28646?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/hooks/useCurrencyRatePolling.test.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  });
+
+  it('should poll currency rates for native currencies when enabled and stop on dismount', async () => {
+    const state = {
+      metamask: {
+        isUnlocked: true,
+        completedOnboarding: true,
+        useCurrencyRateCheck: true,
+        selectedNetworkClientId: 'selectedNetworkClientId',
+        networkConfigurationsByChainId: {
```
Comment (darkwing) [inline ui/hooks/useCurrencyRatePolling.test.ts:32]: Can we add two networks here and ensure it's called twice?

File: ui/hooks/useAccountTrackerPolling.test.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    jest.clearAllMocks();
+  });
+
+  it('should poll account trackers for network client IDs when enabled and stop on dismount', async () => {
+    const state = {
+      metamask: {
+        isUnlocked: true,
+        completedOnboarding: true,
+        selectedNetworkClientId: 'selectedNetworkClientId',
+        networkConfigurationsByChainId: {
```
[the remark this answers is not in the document]
Comment (salimtb) [reply ui/hooks/useAccountTrackerPolling.test.ts:44]: done [5c5b246](https://github.com/MetaMask/metamask-extension/pull/28646/commits/5c5b2468aeda13870ab2db21b6b08ae12e6aea57)

File: ui/hooks/useCurrencyRatePolling.test.ts (hunk 1 above)
Comment (salimtb) [reply ui/hooks/useCurrencyRatePolling.test.ts:32]: done [5c5b246](https://github.com/MetaMask/metamask-extension/pull/28646/commits/5c5b2468aeda13870ab2db21b6b08ae12e6aea57)

File: ui/hooks/useTokenDetectionPolling.test.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    process.env.PORTFOLIO_VIEW = 'true'; // Set your desired mock value here
+
+    mockPromises = [];
+    jest.clearAllMocks();
+  });
+
+  afterEach(() => {
+    // Restore the original value
+    process.env.PORTFOLIO_VIEW = originalPortfolioView;
+  });
```
Comment (gambinish) [inline ui/hooks/useTokenDetectionPolling.test.ts:33]: Will these need to be removed once we nuke the feature flag from the codebase?

File: ui/hooks/useTokenDetectionPolling.test.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+  it('should poll token detection for chain IDs when enabled and stop on dismount', async () => {
+    process.env.PORTFOLIO_VIEW = 'true';
```
Comment (gambinish) [inline ui/hooks/useTokenDetectionPolling.test.ts:36]: Same here

File: ui/hooks/useTokenRatesPolling.test.ts [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    process.env.PORTFOLIO_VIEW = 'true'; // Set your desired mock value here
+
+    mockPromises = [];
+    jest.clearAllMocks();
+  });
+
+  afterEach(() => {
+    // Restore the original value
+    process.env.PORTFOLIO_VIEW = originalPortfolioView;
+  });
```
Comment (gambinish) [inline ui/hooks/useTokenRatesPolling.test.ts:33]: And here

**Review by gambinish — APPROVED** [review]

**Review by sahar-fehri — APPROVED** [review]

**Pull request merged**